service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/PertinentInformationObservationValueMapper.java

```diff
@@ -200,4 +215,137 @@ private static boolean isDenominatorPresent(Ratio ratio) {
+            .collect(Collectors.joining(StringUtils.SPACE));
+
+        if (StringUtils.isNotEmpty(componentText)) {
+            return String.format(COMPONENT_DELIMITER_TEMPLATE, componentText);
+        }
+
+        return componentText;
+    }
+
+    private String extractComponentCode(Observation.ObservationComponentComponent component) {
+        if (component.hasCode()) {
+            var code = CodeableConceptMappingUtils.extractTextOrCoding(component.getCode());
+            if (code.isPresent()) {
+                return String.format(COMPONENT_CODE_TEMPLATE, code.get());
+            }
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String extractComponentValue(Observation.ObservationComponentComponent component) {
+        if (component.hasValue()) {
+            var value = component.getValue();
+
+            return COMPONENT_VALUE_FUNCTIONS.getOrDefault(value.getClass(),
+                $ -> this.mapObservationValueToPertinentInformation(value).trim()
+            ).apply(value);
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String extractComponentInterpretationCode(Observation.ObservationComponentComponent component) {
+        if (component.hasInterpretation() && component.getInterpretation().hasCoding()) {
+            var userSelectedCoding = component
+                .getInterpretation()
+                .getCoding()
+                .stream()
+                .filter(Coding::hasUserSelected)
+                .findFirst();
+
+            var codingToUse = userSelectedCoding.orElseGet(() -> component.getInterpretation().getCodingFirstRep());
+
+            return String.format(COMPONENT_INTERPRETATION_CODE_TEMPLATE, codingToUse.getCode());
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String extractComponentInterpretationText(Observation.ObservationComponentComponent component) {
+        if (component.getInterpretation().hasText()) {
+            return String.format(COMPONENT_INTERPRETATION_TEXT_TEMPLATE, component.getInterpretation().getText());
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String extractComponentReferenceRange(Observation.ObservationComponentComponent component) {
+        if (component.hasReferenceRange()) {
+            var referenceRange = component.getReferenceRangeFirstRep();
+            if (referenceRange.hasLow() && referenceRange.hasHigh()
+                && referenceRange.getLow().hasUnit() && referenceRange.getHigh().hasUnit()) {
+                var lowValue = referenceRange.getLow().getValue().toString();
+                var lowUnit = referenceRange.getLow().getUnit();
+                var highValue = referenceRange.getHigh().getValue().toString();
+                var highUnit = referenceRange.getHigh().getUnit();
+
+                return String.format(COMPONENT_REFERENCE_RANGE_TEMPLATE, lowValue, lowUnit, highValue, highUnit);
+            }
+        }
+        return StringUtils.EMPTY;
+    }
```

Comment:
Can't reference ranges have just a minimum, or just a maximum?

Reply:
It's not explicitly stated in the mapping spreadsheet, will confirm with @dmcavenue1

Reply:
Mapping sheet is slightly abridged for Observation.component because the attributes that can appear in component are the same types that appear elsewhere in Observation - an instance of Observation.component.referenenceRange is identical in structure to Observation.referenceRange and can have both low and high attributes.

Reply:
Thanks for confirming. 
Currently when translating Observation.referenceRange we check that there is low value with unit and high value with unit for it to be translated. Thus believe component.referenceRange performs these same checks here.

Reply:
In which case we need a decision on how to output it when only one is given. Currently:
> "Range: `minVal` `minUnit` `maxVal` `maxUnit`"

e.g.
> "Range: 0.1 mg/L 0.6 mg/L"

How about "Range: min `minVal` `minUnit` max `maxVal` `maxUnit`", where each part can be omitted?
both:
> "Range: min 0.1 mg/L max 0.6 mg/L"

only min:
> "Range: min 0.1 mg/L"

only max:
> "Range: max 0.6 mg/L"

Reply:
Implemented suggested change. 
Reflected in both Observation.component.referenceRange and Observation(uncategorised).value when value is a valueRange - to match mapping spreadsheet.